bin/node/cli/tests/export_import_flow.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+
+use assert_cmd::cargo::cargo_bin;
+use std::{process::Command, fs, path::PathBuf};
+use tempfile::{tempdir, TempDir};
+use regex::Regex;
+use lazy_static::lazy_static;
+
+mod common;
+
+fn contains_error(stderr: &[u8]) -> bool {
+	let log_output = String::from_utf8_lossy(stderr).to_owned();
+    lazy_static! {
+        static ref RE: Regex = Regex::new("Error").unwrap();
+    }
+    RE.is_match(&log_output)
+}
+
+/// Helper struct to execute the export/import/revert tests.
+/// The fields are paths to a temporary directory
+struct ExportImportRevertExecutor<'a> {
+	base_path: &'a TempDir,
+	exported_blocks: &'a PathBuf,
+	db_path: &'a PathBuf,
+}
+
+/// Format options for export / import commands.
+enum FormatOpt {
+	Json,
+	Binary,
+}
+
+impl<'a> ExportImportRevertExecutor<'a> {
+	fn new(base_path: &'a TempDir, exported_blocks: &'a PathBuf, db_path: &'a PathBuf) -> Self {
+		Self {
+			base_path,
+			exported_blocks,
+			db_path,
+		}
+	}
+
+	/// Helper method to run a command.
+	fn run_block_command(&self, command: &str, format_opt: FormatOpt, expected_to_fail: bool) {
+		let arguments = match format_opt {
+			FormatOpt::Binary => vec![command, "--dev", "--pruning", "archive", "--binary", "-d"],
+			FormatOpt::Json => vec![command, "--dev", "--pruning", "archive", "-d"],
+		};
+		let output = Command::new(cargo_bin("substrate"))
+			.args(&arguments)
+			.arg(&self.base_path.path())
+			.arg(&self.exported_blocks)
+			.output()
+			.unwrap();
+
+		if expected_to_fail {
+			// Checking that we did indeed find an error.
+			assert!(contains_error(&output.stderr), "expected to error but did not error!");
+		} else {
+			// Making sure no error were logged.
+			assert!(!contains_error(&output.stderr), "expected not to error but error'd!");
+		}
+
+		// Command should never fail.
+		assert!(output.status.success());
+	}
+
+	/// Runs the `export-blocks` command.
+	fn run_export(&self, fmt_opt: FormatOpt) {
+		self.run_block_command("export-blocks", fmt_opt, false);
+
+		let metadata = fs::metadata(&self.exported_blocks).unwrap();
+		assert!(metadata.len() > 0, "file exported_blocks should not be empty");
+
+		let _ = fs::remove_dir_all(&self.db_path);
```

Comment:
Why delete files manually if `TempDir` is supposed to handle that?

Reply:
So this was [part of the original test]https://github.com/paritytech/substrate/blob/d21c2e5c0fd777a0664ddaf422de10c885872427/bin/node/cli/tests/import_export_and_revert_work.rs#L43) and I believe the main idea is to run the node, export the blocks, and then delete the DB to make sure that `import-blocks` is actually importing blocks, and not "doing nothing".

Maybe what we should add is a test to see that import blocks actually imported blocks? Not sure how to do that, I wouldn't really know what we could query...

Reply:
@arkpar  Yes that was my code actually. First create the temp directory, export, delete the directory, import and then only let the destructor clean-up. It was done on purpose but you could also create another tempdir instead of reusing one, it might make the code more understandable.

Reply:
So keep in mind that now that we've changed the test flow, we need to keep the same directory active for the whole testing period, in order to keep the db :)
I believe the `remove_dir_all` only happens between the `export-blocks` command and the `import-blocks` which makes sense, so that we make sure that we are indeed importing stuff! :)